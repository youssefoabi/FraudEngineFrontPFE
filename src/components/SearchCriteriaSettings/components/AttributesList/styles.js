import { fontSizeBase } from '@cdiscount/ui-fonts';

export default function style() {
  return {
    StyledFormLabel: {
      fontSize: fontSizeBase,
    },
  };
}
