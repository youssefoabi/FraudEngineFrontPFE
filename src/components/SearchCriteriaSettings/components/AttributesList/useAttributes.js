import { find, isEmpty, pathOr, prop, remove, update } from 'ramda';
import { useCallback, useEffect, useMemo, useState } from 'react';

import useParameters from '../ParametersBlock/useParameters';

export default function useAttributes({ attributes, handleChange, engine, currentInputField }) {
  const [key, setKey] = useState('');
  const [externalCodeId, setExternalCodeId] = useState('');

  const { mapParameters } = useParameters({ handleChange });
  const scopeId = pathOr(null, ['currentScope', 'scopeId'], engine);
  const engineParameters = pathOr([], ['parameters'], currentInputField);

  const mappedAttributes = useMemo(() => mapParameters(attributes), [attributes, mapParameters]);

  const onDelete = index => () =>
    handleChange({ id: 'parameters', value: remove(index, 1, engineParameters) });

  const onChangeExternalCodeId = index => e =>
    handleChange({
      id: 'parameters',
      value: update(
        index,
        { ...(attributes[index] || {}), externalCodeId: pathOr('', ['value', 'value', 'externalCode'], e) },
        attributes,
      ),
    });

  const addAttribute = useCallback(() => {
    if (isEmpty(key) || isEmpty(externalCodeId)) return;

    const newItem = {
      key: pathOr(null, ['label'], key),
      externalCodeId: pathOr(null, ['value', 'externalCode'], externalCodeId),
      scopeId,
    };

    const parameter =
      find(
        param =>
          prop('key', newItem) === prop('key', param) &&
          prop('externalCodeId', newItem) === prop('externalCodeId', param) &&
          scopeId === prop('scopeId', param),
        engineParameters,
      ) || null;

    const isNewItem = parameter === null;
    if (!isNewItem) return;

    handleChange({ id: 'parameters', value: [...engineParameters, newItem] });
    setKey('');
    setExternalCodeId('');
  }, [key, externalCodeId, scopeId, engineParameters, handleChange]);

  useEffect(() => {
    addAttribute();
  }, [key, externalCodeId, addAttribute]);

  const onHandleKey = e => setKey(prop('value', e));
  const onHandleExternalKey = e => setExternalCodeId(prop('value', e));

  const onHandleAttribute = index => e =>
    handleChange({
      id: 'parameters',
      value: update(
        index,
        { ...attributes[index], key: pathOr('', ['value', 'value', 'key'], e) },
        attributes,
      ),
    });

  return {
    mappedAttributes,
    key,
    externalCodeId,
    addAttribute,
    onChangeExternalCodeId,
    onDelete,
    onHandleKey,
    onHandleExternalKey,
    onHandleAttribute,
  };
}
