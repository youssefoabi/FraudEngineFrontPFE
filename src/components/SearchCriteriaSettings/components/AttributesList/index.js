import { AddCircle, RemoveCircle } from '@material-ui/icons';
import { arrayOf, func, shape } from 'prop-types';
import { Button, Field, GridItem, useStyles } from '@cdiscount/backoffice-ui';
import { ceruleano, red } from '@cdiscount/ui-colors';
import { FormLabel } from '@material-ui/core';
import { isNil, pathOr } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';
import Grid from '@material-ui/core/Grid';
import React from 'react';

import { enginePropTypes } from '../../../EngineForm/enginePropTypes';
import { ExternalCodeByIds, ExternalCodeIdsOptions, KeysOptions } from './config';
import styles from './styles';
import useAttributes from './useAttributes';

const AttributesList = ({ attributes, handleChange, allParameters, engine, currentInputField }) => {
  const {
    onDelete,
    mappedAttributes,
    key,
    externalCodeId,
    addAttribute,
    onChangeExternalCodeId,
    onHandleKey,
    onHandleExternalKey,
    onHandleAttribute,
  } = useAttributes({ handleChange, attributes, engine, currentInputField });

  const renderRow = (option, index) => {
    const isAddBloc = isNil(index) && index !== 0;
    const addedExternalId = pathOr(null, ['value', 'externalCodeId'], option);

    if (!isAddBloc && isNilOrEmpty(option)) return null;

    return (
      <Grid container spacing={3}>
        <Grid item columns={5} xs={5} md={5} lg={5}>
          <Field
            label="Attribut"
            type="select"
            onChange={isAddBloc ? onHandleExternalKey : onChangeExternalCodeId(index)}
            value={isAddBloc ? externalCodeId : pathOr({}, [addedExternalId], ExternalCodeByIds)}
            selectProps={ExternalCodeIdsOptions}
          />
        </Grid>
        <Grid item columns={5} xs={5} md={5} lg={5}>
          <Field
            label="Variable Correspondante"
            type="select"
            onChange={isAddBloc ? onHandleKey : onHandleAttribute(index)}
            value={isAddBloc ? key : option}
            selectProps={KeysOptions(allParameters)}
          />
        </Grid>
        <GridItem xs={1} md={1} lg={1}>
          {isAddBloc ? (
            <Button size="large" icon={AddCircle} onClick={addAttribute} style={{ color: ceruleano }} />
          ) : (
            <Button size="large" icon={RemoveCircle} onClick={onDelete(index)} style={{ color: red }} />
          )}
        </GridItem>
      </Grid>
    );
  };

  const renderAttributesList = () => {
    if (isNilOrEmpty(mappedAttributes)) return null;

    return mappedAttributes.map(renderRow);
  };

  const classes = useStyles(styles);

  return (
    <>
      <FormLabel className={classes.StyledFormLabel}>Parametre lie dans l&apos;url de recherche</FormLabel>
      <div>
        <br />
        {renderAttributesList()}
        {renderRow()}
      </div>
    </>
  );
};

AttributesList.propTypes = {
  allParameters: arrayOf(shape({})),
  attributes: arrayOf(shape({})),
  handleChange: func,
  engine: enginePropTypes,
  currentInputField: shape({}),
};

AttributesList.defaultProps = {
  allParameters: null,
  attributes: null,
  handleChange: Function.Prototype,
  engine: null,
  currentInputField: null,
};

export default AttributesList;
