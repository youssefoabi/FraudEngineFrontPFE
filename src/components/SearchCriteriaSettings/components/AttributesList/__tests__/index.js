import { itRendersAllMutations } from '@cdiscount/jest-utils';

import AttributesList from '../';

jest.resetModules();

jest.mock('../useAttributes', () => ({
  __esModule: true,
  default: jest
    .fn()
    .mockReturnValueOnce({})
    .mockReturnValueOnce({
      mappedAttributes: [],
    })
    .mockReturnValueOnce({
      mappedAttributes: null,
    })
    .mockReturnValueOnce({
      mappedAttributes: [
        { label: 'search', value: { id: 3, scopeId: 2, key: 'search', externalCodeId: 1 } },
        { label: 'cp', value: { id: 0, scopeId: 1, key: 'cp', externalCodeId: 2 } },
        { label: 'commune', value: { id: 1, scopeId: 1, key: 'commune', externalCodeId: 4 } },
      ],
      onChangeExternalCodeId: jest.fn(),
      onHandleAttribute: jest.fn(),
      onDelete: jest.fn(),
      addAttribute: jest.fn(),
      onHandleKey: jest.fn(),
      onHandleExternalKey: jest.fn(),
    }),
  namedExport: 'useAttributes',
}));

const mutations = [
  { name: 'without props' },
  { name: 'with empty mappedAttributes' },
  { name: 'with null mapedAttribute' },
  { name: 'with mapedAttribute' },
];

describe(AttributesList.name, () => {
  itRendersAllMutations(AttributesList, mutations);
});
