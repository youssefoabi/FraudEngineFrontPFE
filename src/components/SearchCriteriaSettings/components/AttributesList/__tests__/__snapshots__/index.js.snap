// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributesList AttributesList renders correctly with empty mappedAttributes 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormLabel))
    className="makeStyles-StyledFormLabel-2"
  >
    Parametre lie dans l'url de recherche
  </WithStyles(ForwardRef(FormLabel))>
  <div>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Attribut"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": Array [
                Object {
                  "label": "Commune",
                  "value": Object {
                    "externalCode": 1,
                  },
                },
                Object {
                  "label": "Code postal",
                  "value": Object {
                    "externalCode": 2,
                  },
                },
                Object {
                  "label": "Latitude",
                  "value": Object {
                    "externalCode": 3,
                  },
                },
                Object {
                  "label": "Longitude",
                  "value": Object {
                    "externalCode": 4,
                  },
                },
              ],
              "placeholder": "Attribut",
            }
          }
          size="small"
          type="select"
          value=""
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Variable Correspondante"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": null,
              "placeholder": "Variable correspondante",
            }
          }
          size="small"
          type="select"
          value=""
        />
      </WithStyles(ForwardRef(Grid))>
      <h
        fullheight={false}
        lg={1}
        md={1}
        title={null}
        xs={1}
      >
        <y
          color="primary"
          icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "AddCircleIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            }
          }
          label={null}
          size="large"
          style={
            Object {
              "color": "#00adee",
            }
          }
        />
      </h>
    </WithStyles(ForwardRef(Grid))>
  </div>
</Fragment>
`;

exports[`AttributesList AttributesList renders correctly with mapedAttribute 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormLabel))
    className="makeStyles-StyledFormLabel-4"
  >
    Parametre lie dans l'url de recherche
  </WithStyles(ForwardRef(FormLabel))>
  <div>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Attribut"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": Array [
                Object {
                  "label": "Commune",
                  "value": Object {
                    "externalCode": 1,
                  },
                },
                Object {
                  "label": "Code postal",
                  "value": Object {
                    "externalCode": 2,
                  },
                },
                Object {
                  "label": "Latitude",
                  "value": Object {
                    "externalCode": 3,
                  },
                },
                Object {
                  "label": "Longitude",
                  "value": Object {
                    "externalCode": 4,
                  },
                },
              ],
              "placeholder": "Attribut",
            }
          }
          size="small"
          type="select"
          value={
            Object {
              "label": "Commune",
              "value": Object {
                "externalCode": 1,
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Variable Correspondante"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": null,
              "placeholder": "Variable correspondante",
            }
          }
          size="small"
          type="select"
          value={
            Object {
              "label": "search",
              "value": Object {
                "externalCodeId": 1,
                "id": 3,
                "key": "search",
                "scopeId": 2,
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <h
        fullheight={false}
        lg={1}
        md={1}
        title={null}
        xs={1}
      >
        <y
          color="primary"
          icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "RemoveCircleIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            }
          }
          label={null}
          size="large"
          style={
            Object {
              "color": "#f00",
            }
          }
        />
      </h>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Attribut"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": Array [
                Object {
                  "label": "Commune",
                  "value": Object {
                    "externalCode": 1,
                  },
                },
                Object {
                  "label": "Code postal",
                  "value": Object {
                    "externalCode": 2,
                  },
                },
                Object {
                  "label": "Latitude",
                  "value": Object {
                    "externalCode": 3,
                  },
                },
                Object {
                  "label": "Longitude",
                  "value": Object {
                    "externalCode": 4,
                  },
                },
              ],
              "placeholder": "Attribut",
            }
          }
          size="small"
          type="select"
          value={
            Object {
              "label": "Code postal",
              "value": Object {
                "externalCode": 2,
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Variable Correspondante"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": null,
              "placeholder": "Variable correspondante",
            }
          }
          size="small"
          type="select"
          value={
            Object {
              "label": "cp",
              "value": Object {
                "externalCodeId": 2,
                "id": 0,
                "key": "cp",
                "scopeId": 1,
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <h
        fullheight={false}
        lg={1}
        md={1}
        title={null}
        xs={1}
      >
        <y
          color="primary"
          icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "RemoveCircleIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            }
          }
          label={null}
          size="large"
          style={
            Object {
              "color": "#f00",
            }
          }
        />
      </h>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Attribut"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": Array [
                Object {
                  "label": "Commune",
                  "value": Object {
                    "externalCode": 1,
                  },
                },
                Object {
                  "label": "Code postal",
                  "value": Object {
                    "externalCode": 2,
                  },
                },
                Object {
                  "label": "Latitude",
                  "value": Object {
                    "externalCode": 3,
                  },
                },
                Object {
                  "label": "Longitude",
                  "value": Object {
                    "externalCode": 4,
                  },
                },
              ],
              "placeholder": "Attribut",
            }
          }
          size="small"
          type="select"
          value={
            Object {
              "label": "Longitude",
              "value": Object {
                "externalCode": 4,
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Variable Correspondante"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": null,
              "placeholder": "Variable correspondante",
            }
          }
          size="small"
          type="select"
          value={
            Object {
              "label": "commune",
              "value": Object {
                "externalCodeId": 4,
                "id": 1,
                "key": "commune",
                "scopeId": 1,
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <h
        fullheight={false}
        lg={1}
        md={1}
        title={null}
        xs={1}
      >
        <y
          color="primary"
          icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "RemoveCircleIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            }
          }
          label={null}
          size="large"
          style={
            Object {
              "color": "#f00",
            }
          }
        />
      </h>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Attribut"
          multiline={false}
          onChange={[MockFunction]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": Array [
                Object {
                  "label": "Commune",
                  "value": Object {
                    "externalCode": 1,
                  },
                },
                Object {
                  "label": "Code postal",
                  "value": Object {
                    "externalCode": 2,
                  },
                },
                Object {
                  "label": "Latitude",
                  "value": Object {
                    "externalCode": 3,
                  },
                },
                Object {
                  "label": "Longitude",
                  "value": Object {
                    "externalCode": 4,
                  },
                },
              ],
              "placeholder": "Attribut",
            }
          }
          size="small"
          type="select"
          value=""
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Variable Correspondante"
          multiline={false}
          onChange={[MockFunction]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": null,
              "placeholder": "Variable correspondante",
            }
          }
          size="small"
          type="select"
          value=""
        />
      </WithStyles(ForwardRef(Grid))>
      <h
        fullheight={false}
        lg={1}
        md={1}
        title={null}
        xs={1}
      >
        <y
          color="primary"
          icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "AddCircleIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            }
          }
          label={null}
          onClick={[MockFunction]}
          size="large"
          style={
            Object {
              "color": "#00adee",
            }
          }
        />
      </h>
    </WithStyles(ForwardRef(Grid))>
  </div>
</Fragment>
`;

exports[`AttributesList AttributesList renders correctly with null mapedAttribute 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormLabel))
    className="makeStyles-StyledFormLabel-3"
  >
    Parametre lie dans l'url de recherche
  </WithStyles(ForwardRef(FormLabel))>
  <div>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Attribut"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": Array [
                Object {
                  "label": "Commune",
                  "value": Object {
                    "externalCode": 1,
                  },
                },
                Object {
                  "label": "Code postal",
                  "value": Object {
                    "externalCode": 2,
                  },
                },
                Object {
                  "label": "Latitude",
                  "value": Object {
                    "externalCode": 3,
                  },
                },
                Object {
                  "label": "Longitude",
                  "value": Object {
                    "externalCode": 4,
                  },
                },
              ],
              "placeholder": "Attribut",
            }
          }
          size="small"
          type="select"
          value=""
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Variable Correspondante"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": null,
              "placeholder": "Variable correspondante",
            }
          }
          size="small"
          type="select"
          value=""
        />
      </WithStyles(ForwardRef(Grid))>
      <h
        fullheight={false}
        lg={1}
        md={1}
        title={null}
        xs={1}
      >
        <y
          color="primary"
          icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "AddCircleIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            }
          }
          label={null}
          size="large"
          style={
            Object {
              "color": "#00adee",
            }
          }
        />
      </h>
    </WithStyles(ForwardRef(Grid))>
  </div>
</Fragment>
`;

exports[`AttributesList AttributesList renders correctly without props 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormLabel))
    className="makeStyles-StyledFormLabel-1"
  >
    Parametre lie dans l'url de recherche
  </WithStyles(ForwardRef(FormLabel))>
  <div>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Attribut"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": Array [
                Object {
                  "label": "Commune",
                  "value": Object {
                    "externalCode": 1,
                  },
                },
                Object {
                  "label": "Code postal",
                  "value": Object {
                    "externalCode": 2,
                  },
                },
                Object {
                  "label": "Latitude",
                  "value": Object {
                    "externalCode": 3,
                  },
                },
                Object {
                  "label": "Longitude",
                  "value": Object {
                    "externalCode": 4,
                  },
                },
              ],
              "placeholder": "Attribut",
            }
          }
          size="small"
          type="select"
          value=""
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        columns={5}
        item={true}
        lg={5}
        md={5}
        xs={5}
      >
        <G
          alt={null}
          disabled={false}
          icon={null}
          id={null}
          imageSize="10x10"
          inputType="text"
          label="Variable Correspondante"
          multiline={false}
          onChange={[Function]}
          options={null}
          order={null}
          placeholder={null}
          selectProps={
            Object {
              "options": null,
              "placeholder": "Variable correspondante",
            }
          }
          size="small"
          type="select"
          value=""
        />
      </WithStyles(ForwardRef(Grid))>
      <h
        fullheight={false}
        lg={1}
        md={1}
        title={null}
        xs={1}
      >
        <y
          color="primary"
          icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "AddCircleIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            }
          }
          label={null}
          size="large"
          style={
            Object {
              "color": "#00adee",
            }
          }
        />
      </h>
    </WithStyles(ForwardRef(Grid))>
  </div>
</Fragment>
`;
